constantine/serialization/codecs_banderwagon.nim

```diff
@@ -110,4 +110,44 @@ func deserialize*(dst: var EC_Prj, src: array[32, byte]): CttCodecEccStatus =
   if not(bool dst.isInSubgroup()):
     return cttCodecEcc_PointNotInSubgroup
 
-  return cttCodecEcc_Success
\ No newline at end of file
+  return cttCodecEcc_Success
+
+## ############################################################
+##
+##              Banderwagon Batch Serialization
+##
+## ############################################################
+
+func serializeBatch*(
+    dst: ptr UncheckedArray[array[32, byte]],
+    points: ptr UncheckedArray[EC_Prj],
+    N: static int,
+  ) : CttCodecEccStatus =
+
+  # collect all the z coordinates
+  var zs: array[N, Fp[Banderwagon]]
+  var zs_inv: array[N, Fp[Banderwagon]]
```

Comment:
if N is known at runtime, then we have to use the `allocStackArray`.
Additionally tag the function as `{.noInline.}` to ensure stack is properly freed.